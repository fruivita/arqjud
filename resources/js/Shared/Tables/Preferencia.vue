<!--
    Container para as preferências de visualização da tabelas.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import Container from '@/Shared/Containers/Container.vue';
import Icone from '@/Shared/Icones/Icone.vue';
import PorPagina from '@/Shared/Tables/PorPagina.vue';
import { onClickOutside } from '@vueuse/core';
import { ref } from 'vue';

const visivel = ref(false);

const container = ref(null);

onClickOutside(container, () => (visivel.value = false));
</script>

<template>
    <div class="flex justify-end">
        <div class="inline-flex flex-col text-right" ref="container">
            <div>
                <button
                    @click="visivel = !visivel"
                    class="transform-gpu opacity-50 transition duration-300 hover:opacity-100"
                    dusk="toggle"
                >
                    <Icone nome="three-dots-vertical" />
                </button>
            </div>

            <Transition
                enter-from-class="opacity-0 scale-50"
                enter-to-class="opacity-100 scale-100"
                enter-active-class="transition duration-300 transform-gpu"
                leave-active-class="transition duration-200 transform-gpu"
                leave-from-class="opacity-100 scale-100"
                leave-to-class="opacity-0 scale-50"
            >
                <Container v-show="visivel" dusk="preferencias">
                    <div class="inline-flex">
                        <div class="space-y-3">
                            <div class="grid grid-cols-1 gap-3 md:grid-cols-2" dusk="slot">
                                <slot></slot>
                            </div>

                            <PorPagina />
                        </div>
                    </div>
                </Container>
            </Transition>
        </div>
    </div>
</template>
