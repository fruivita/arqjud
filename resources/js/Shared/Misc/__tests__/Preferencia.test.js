/**
 * Testes para o componente Preferencia.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Preferencia from '@/Shared/Misc/Preferencia.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Preferencia, { ...options });
    };
});

// Caminho feliz
describe('Preferencia', () => {
    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { default: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('exibe o conteúdo escondido ao clicar no botão respeitando o snapshot', async () => {
        const wrapper = mountFunction({
            slots: { default: 'foo' },
        });

        await wrapper.find('[dusk="toggle"]').trigger('click');

        expect(wrapper.html()).toMatchSnapshot();
    });

    test('alterna a visibilidade do conteúdo do componente ao clicar no botão', async () => {
        const wrapper = mountFunction({
            slots: { default: 'foo' },
        });

        const preferencias = wrapper.get('[dusk="preferencias"]');

        expect(preferencias.isVisible()).toBeFalsy();

        await wrapper.get('[dusk="toggle"]').trigger('click');

        expect(preferencias.isVisible()).toBeTruthy();

        await wrapper.get('[dusk="toggle"]').trigger('click');

        expect(preferencias.isVisible()).toBeFalsy();
    });
});
