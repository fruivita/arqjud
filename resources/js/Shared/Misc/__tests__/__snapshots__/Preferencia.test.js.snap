// Vitest Snapshot v1

exports[`Preferencia > exibe o conteúdo escondido ao clicar no botão respeitando o snapshot 1`] = `
"<div class=\\"flex justify-end\\">
  <div class=\\"inline-flex flex-col text-right\\">
    <div><button class=\\"transform-gpu opacity-50 transition duration-300 hover:opacity-100\\" dusk=\\"toggle\\"><svg class=\\"w-6 h-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
          <path d=\\"M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z\\"></path>
        </svg></button></div>
    <transition-stub enterfromclass=\\"opacity-0 scale-50\\" entertoclass=\\"opacity-100 scale-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100 scale-100\\" leavetoclass=\\"opacity-0 scale-50\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
      <div class=\\"overflow-x-auto rounded-lg p-3 shadow-lg shadow-secundaria-500 dark:shadow-primaria-500\\" dusk=\\"preferencias\\" style=\\"\\">
        <div class=\\"inline-flex\\">
          <div class=\\"space-y-3\\">
            <div class=\\"grid grid-cols-1 gap-3 md:grid-cols-2\\" dusk=\\"slot\\">foo</div>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Preferencia > renderiza o componente respeitando o snapshot 1`] = `
"<div class=\\"flex justify-end\\">
  <div class=\\"inline-flex flex-col text-right\\">
    <div><button class=\\"transform-gpu opacity-50 transition duration-300 hover:opacity-100\\" dusk=\\"toggle\\"><svg class=\\"w-6 h-6\\" fill=\\"currentColor\\" viewBox=\\"0 0 16 16\\" xmlns=\\"http://www.w3.org/2000/svg\\">
          <path d=\\"M9.5 13a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0zm0-5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0z\\"></path>
        </svg></button></div>
    <transition-stub enterfromclass=\\"opacity-0 scale-50\\" entertoclass=\\"opacity-100 scale-100\\" enteractiveclass=\\"transition duration-300 transform-gpu\\" leaveactiveclass=\\"transition duration-200 transform-gpu\\" leavefromclass=\\"opacity-100 scale-100\\" leavetoclass=\\"opacity-0 scale-50\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
      <div class=\\"overflow-x-auto rounded-lg p-3 shadow-lg shadow-secundaria-500 dark:shadow-primaria-500\\" dusk=\\"preferencias\\" style=\\"display: none;\\">
        <div class=\\"inline-flex\\">
          <div class=\\"space-y-3\\">
            <div class=\\"grid grid-cols-1 gap-3 md:grid-cols-2\\" dusk=\\"slot\\">foo</div>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;
